src/main/java/com/programmers/librarymanagement/repository/NormalBookRepository.java

```diff
@@ -0,0 +1,77 @@
+package com.programmers.librarymanagement.repository;
+
+import com.programmers.librarymanagement.domain.Book;
+import com.programmers.librarymanagement.domain.Status;
+import com.programmers.librarymanagement.utils.CsvFileIo;
+
+import java.time.LocalDateTime;
+import java.util.*;
+
+public class NormalBookRepository implements BookRepository {
+
+    private final CsvFileIo csvFileIo = new CsvFileIo();
+    private final Map<Long, Book> libraryStorage = addBookByFile(csvFileIo.readCsv());
+
+    @Override
+    public void addBook(Book book) {
+
+        libraryStorage.put(book.getId(), book);
+        csvFileIo.writeCsv(book); // 콘솔로 등록한 도서 정보를 파일에도 저장
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        return Optional.ofNullable(libraryStorage.get(id));
+    }
+
+    @Override
+    public List<Book> findByTitle(String title) {
+        return libraryStorage.values().stream()
+                .filter(book -> book.isContain(title))
+                .toList();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return new ArrayList<>(libraryStorage.values());
```

Comment:
조회 용도로 새로운 리스트로 감쌀 거라면 그 리스트에 대해 변경가능성이 최대한 적게 Immutable로 만드는 것이 좋습니다.

```suggestion
        return libraryStorage.values()
                .stream()
                .toList();
```

Reply:
mutable, immutable까지는 생각하지 못 했습니다... 수정하도록 하겠습니다!